modules/gdscript/gdscript_analyzer.cpp

```diff
@@ -3701,11 +3701,16 @@ void GDScriptAnalyzer::const_fold_dictionary(GDScriptParser::DictionaryNode *p_d
 GDScriptParser::DataType GDScriptAnalyzer::type_from_variant(const Variant &p_value, const GDScriptParser::Node *p_source) {
 	GDScriptParser::DataType result;
 	result.is_constant = true;
-	result.kind = GDScriptParser::DataType::BUILTIN;
 	result.builtin_type = p_value.get_type();
 	result.type_source = GDScriptParser::DataType::ANNOTATED_EXPLICIT; // Constant has explicit type.
 
-	if (p_value.get_type() == Variant::OBJECT) {
+	if (p_value.get_type() == Variant::NIL) {
+		// A null value is a variant, not void.
```

Comment:
Is this always true? I don't know what `p_value` represents here exactly but in PropertyInfo, there's `PROPERTY_USAGE_NIL_IS_VARIANT` for the case where `Variant::NIL` should be handled as `Variant()`. Otherwise it's indeed `void`.

Reply:
Notably, what happens with this PR and this MRP?
```
extends Node

func get_param():
	return

func _ready():
	var test = get_param()
	print(test)
```

Reply:
this function is for when the analyzer is given a variant and needs to make a corresponding datatype, 
as far as i know, void is only used in specific places and is never inferred from a value
in all the places this function is used, i wouldn't expect void to be returned
and the func in that code snippet would be void, because there's no null literal to pass to `type_from_variant`
https://github.com/godotengine/godot/blob/daf168f4c821980a6a06fde67057f6a14d9e20c6/modules/gdscript/gdscript_analyzer.cpp#L1781-L1785